spec/views/posts/user_post_index_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe 'user_posts/#index', type: :feature do
+  before :each do
+    @user = User.first
+    visit user_posts_path(@user.id)
+  end
+  describe 'Users Posts index view' do
+    it 'should show user profile picture' do
+      expect(page).to have_css("img[src*='#{@user.photo}']")
+    end
+    it 'should show user username' do
+      expect(page).to have_content(@user.name)
+    end
+    it 'should show the number of posts the user have written' do
+      expect(page).to have_content("Number of posts: #{@user.post_counter}")
+    end
+    it "should show the post's title" do
+      @user.posts.each { |post| expect(page).to have_content(post.title) }
+    end
+    it "should show the post's body" do
+      @user.posts.each { |post| expect(page).to have_content(post.text) }
+    end
+    it "should show the post's first comment" do
+      first_comment = @user.posts.first
+      expect(page).to have_content(first_comment.text)
+    end
+    it 'should show how many comments a post has' do
+      post = @user.posts.first
+      expect(page).to have_content("Comments: #{post.comments_counter}")
+    end
+    it 'should have button for pagination' do
+      expect(page).to have_button('Pagination')
+    end
+  end
+end
```

Comment:
- [ ] You might have forgotten to add the tests for "I can see how many likes a post has" Kindly include it.

![image](https://user-images.githubusercontent.com/86318284/217479292-50542ee8-efb3-4410-9674-6781638383c9.png)

- [ ] You might have forgotten to add the tests for "When I click on a post, it redirects me to that post's show page." Kindly include it.

![image](https://user-images.githubusercontent.com/86318284/217479670-eb55a235-5563-4bfe-94fa-141b7a7a087b.png)

Reply:
I have added the 2 tests.

Please note that all tests are passing here. 

<img width="567" alt="Screenshot 2023-02-08 at 12 35 19 PM" src="https://user-images.githubusercontent.com/48407831/217518908-67840af5-7fae-4383-a087-8d944e79756a.png">